Chummer/data/metamagic.xsd

```diff
@@ -1,58 +1,52 @@
-<?xml version="1.0" encoding="utf-8"?>
-<xs:schema id="NewDataSet" xmlns="" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:msdata="urn:schemas-microsoft-com:xml-msdata">
-<xs:include schemaLocation="bonuses.xsd" />
+﻿<?xml version='1.0' encoding='UTF-8'?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:vc="http://www.w3.org/2007/XMLSchema-versioning" elementFormDefault="qualified" vc:minVersion="1.1">
+  <xs:include schemaLocation="bonuses.xsd" />
   <xs:include schemaLocation="conditions.xsd" />
+  <xs:include schemaLocation="commons.xsd" />
+
   <xs:element name="chummer">
     <xs:complexType>
       <xs:sequence>
-        <xs:element name="version" type="xs:string" minOccurs="0" />
-        <xs:element name="metamagics" minOccurs="0" maxOccurs="unbounded">
+
+        <xs:element name="metamagics">
           <xs:complexType>
             <xs:sequence>
-              <xs:element name="metamagic" minOccurs="0" maxOccurs="unbounded">
+              <xs:element name="metamagic" maxOccurs="unbounded">
                 <xs:complexType>
-                  <xs:sequence>
-                    <xs:element name="id" type="xs:string" minOccurs="1" maxOccurs="1" />
-                    <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
-                    <xs:element name="hide" type="xs:string" minOccurs="0" maxOccurs="1" />
-                    <xs:element name="adept" type="xs:string" minOccurs="1" maxOccurs="1" />
-                    <xs:element name="magician" type="xs:string" minOccurs="1" maxOccurs="1" />
-                    <xs:element name="limit" type="xs:string" minOccurs="0" />
-                    <xs:element name="bonus" type="bonusTypes"  minOccurs="0" maxOccurs="1" />
-                    <xs:element ref="forbidden" minOccurs="0" maxOccurs="1" />
-                    <xs:element ref="required" minOccurs="0" maxOccurs="1" />
-                    <xs:element name="source" type="xs:string" minOccurs="0" />
-                    <xs:element name="page" type="xs:string" minOccurs="0" />
-                  </xs:sequence>
+                  <xs:all>
+                    <xs:group ref="baseGroup" />
+
+                    <!-- required -->
+                    <xs:element name="adept" type="boolean" />
+                    <xs:element name="magician" type="boolean" />
+
+                    <!-- optional -->
+                    <xs:element ref="bonus" minOccurs="0" />
+                    <xs:element name="required" minOccurs="0" />
```

Comment:
This should be a ref to refer to conditions.xsd, shouldn't it?

Reply:
You are corrected, I fixed it on tradition.xsd as well. Unfortunately, xerces assume any type when you don't specify, and it might be overseed easily.